src/shogun/mathematics/Seedable.h

```diff
@@ -45,12 +45,19 @@ namespace shogun
 			return "Seedable";
 		}
 
+		bool is_set_seed() const
```

Comment:
I am not a fan of this tbh

Reply:
I am thinking if you would expect the seed to not be propagated to the subobjects.. In crossvalidation we always propagate the seed right now.. If you want to have different seeds you need to either set the crossvalidation seed before adding the machine, or set the seed of the machine after propagating the seed from crossvalidation.

Reply:
in sklearn it seems like each object's seed only affects it's own rng, e.g. KFolds: "random_state affects the ordering of the indices, which controls the randomness of each fold. Otherwise, this parameter has no effect"

Reply:
But then again, in sklearn you can set the random state of numpy which will affect all the rng. I think we don't have an equivalent

Reply:
nope we don't. I think that is why we set up this recursive seed passing? Personally, if I set the seed in a machine before training, I would expect that the results of say "train" would be deterministic...